tests/fixtures/make_partial_repo.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+set -eu -o pipefail
+
+mkdir base
+(cd base
+    git init -q
+    git checkout -b main
+    touch code.rs
+    git add code.rs
+    git commit -q -m c1
+    echo hello >> code.rs
+    git add code.rs
+    git commit -q -m c2
+    echo world >> code.rs
+    git add code.rs
+    git commit -q -m c3
+    echo something >> code.rs
+    git add code.rs
+    git commit -q -m c4
+    echo more >> code.rs
+    git mv code.rs renamed.rs
+    echo change >> renamed.rs
+    git commit -q -am c5
+
+    git config uploadpack.allowfilter true
```

Comment:
This makes `--filter=blob:none` work, otherwise we see a warning when cloning, saying that the option is ignored as the server didn't support it.